fast_response/FastResponseAnalysis.py

```diff
@@ -992,6 +1020,424 @@ def unblind_TS(self, custom_events=None):
+        #Write to healpy
+        outputProbHPMap=np.zeros(12*nside**2)
+        ras_graph,decs_graph=zip(*list(finalProbs.keys()))
+        outputProbHPMap[hp.ang2pix(nside=nside,theta=np.array(ras_graph)*180/np.pi, phi=np.array(decs_graph)*180/np.pi,lonlat=True)]=np.exp(np.array(list(finalProbs.values())))
+        
+        t2 = time.time()
+        print("finished posterior calc, took {} s".format(t2-t1))
+
+        #set 0s to nan for plotting 
+        outputProbHPMapPlotting=copy.deepcopy(outputProbHPMap)
+        outputProbHPMapPlotting[outputProbHPMapPlotting == 0] = np.nan
+
+        #Add Plotting
+        if (plotting_location is not None):
+            fig, ax=plt.subplots(1,1,figsize=(10,10))
+            ras_graph,decs_graph=zip(*list(finalProbs.keys()))
+            fig, ax=plt.subplots(1,1,figsize=(10,10))
+            plt.sca(ax)
+            reso=.1
+            sizeZoomInDegs=10
+            nPixZoom=sizeZoomInDegs/reso*60
+            
+            
+            hp.visufunc.mollview(map=outputProbHPMapPlotting,hold=True,title="Posterior Skymap",
+                                 bgcolor="white",badcolor="white",rot=(180,0,0),)
+            hp.graticule(verbose=True)
+            ax=plt.gca()
+            text_objs = ax.findobj(
+                lambda artist: isinstance(artist, Text)
+            )
+            for obj in text_objs:
+                obj.set_fontsize(30)
+            plt.savefig(plotting_location+"PosteriorMap.png")
+            plt.close()
+
+            #plot zoomed 
+            fig, ax=plt.subplots(1,1,figsize=(10,10))
+            plt.sca(ax)
+            hp.visufunc.gnomview(map=outputProbHPMapPlotting,hold=True,title="Posterior Skymap",rot=(ramax*180/np.pi,decmax*180/np.pi,0),
+                xsize=nPixZoom,ysize=nPixZoom,reso=reso,bgcolor="white",badcolor="grey",margins=(.02,.02,.05,0),notext=True,cbar=False)
+            hp.graticule(verbose=True)
+            plotting_utils.plot_labels(decmax,ramax, reso,nPix=nPixZoom,label_scale=.97)
+            ax=plt.gca()
+            text_objs = ax.findobj(
+                lambda artist: isinstance(artist, Text)
+            )
+            for obj in text_objs:
+                obj.set_fontsize(30)
+            plotting_utils.plot_color_bar(range=[0,np.nanmax(outputProbHPMapPlotting)], cmap="viridis", col_label=r"Posterior PDF",
+                    offset=-40,labels=[0,"{0:.1e}".format(np.nanmax(outputProbHPMapPlotting)/2),"{0:.2e}".format(np.nanmax(outputProbHPMapPlotting))],loc=[0.86, 0.2, 0.03, 0.6])
+            theta, phi =plotting_utils.plot_contours([.9],outputProbHPMap)
+            hp.projplot(theta[0], phi[0], linewidth=2., c='k')
+            for i in range(1, len(theta)):
+                hp.projplot(theta[i], phi[i], linewidth=2., c='k', label=None)
+            plt.savefig(plotting_location+"PosteriorMapZoomed.png")
+            plt.close()
+
+            
+            #plot zoomed with events 
```

Comment:
a lot of this repeats. can you clean this up by separating parts that are used multiple times into their own helper functions in `plotting_utils`?